bash-preexec.sh

```diff
@@ -319,10 +345,13 @@ __bp_install() {
     fi;
     PROMPT_COMMAND+='__bp_interactive_mode'
 
-    # Add two functions to our arrays for convenience
-    # of definition.
-    precmd_functions+=(precmd)
-    preexec_functions+=(preexec)
+    # Add two functions to our arrays for convenience of definition only when
+    # the functions have not yet added.
+    if [[ ! ${__bp_installed_convenience_functions-} ]]; then
```

Comment:
why is this check necessary? Are there instances where `__bp_install` is being called twice?

Reply:
Yes, when the user or another library would like to temporarily turn off bash-preexec and then later want to re-enable it. `ble.sh` would do that.

Reply:
Curious, what's the reason ble.sh would re-enable it?

Reply:
ble.sh allows a user to temporarily turn off ble.sh by `ble-detach`. (For example, I have received some reports from a user that there is a performance issue with ble.sh in a directory with tens of thousands of files so the user temporarily turns off ble.sh while the current working directory is in that directory.) In the detached state, ble.sh would like to re-enable the original `bash-preexec.sh` hook because ble.sh does not provide the feature in the detached state.

Reply:
Can we add a small test to cover this?

Reply:
Yes! Thank you for the suggestion! I have added the test in commit e846228.